src/core/core.scale.js

```diff
@@ -419,8 +419,11 @@ module.exports = Element.extend({
 			}
 		}
 
-		// Don't bother fitting the ticks if we are not showing them
-		if (tickOpts.display && display) {
+		// Don't bother fitting the ticks if we are not showing the labels
+		if (!tickOpts.display || !display) {
+			me._maxLabelLines = 0;
```

Comment:
I think these should be initialized [here](https://github.com/chartjs/Chart.js/blob/719ab0d76860986a71f7f887ff30a69f45bde370/src/core/core.scale.js#L175) unconditionally `0`

I'd also add a `|| 0` to https://github.com/chartjs/Chart.js/blob/719ab0d76860986a71f7f887ff30a69f45bde370/src/core/core.scale.js#L685 and https://github.com/chartjs/Chart.js/blob/719ab0d76860986a71f7f887ff30a69f45bde370/src/core/core.scale.js#L689 to be certain we don't get `NaN`.

Reply:
Done